Realm/Realm/Handles/SessionHandle.cs

```diff
@@ -201,35 +201,38 @@ public void ShutdownAndWait()
         [MonoPInvokeCallback(typeof(NativeMethods.SessionErrorCallback))]
         private static void HandleSessionError(IntPtr sessionHandlePtr, ErrorCode errorCode, PrimitiveValue message, IntPtr userInfoPairs, IntPtr userInfoPairsLength, bool isClientReset)
         {
-            try
-            {
-                using var handle = new SessionHandle(null, sessionHandlePtr);
-                var session = new Session(handle);
-                var messageString = message.AsString();
+            var handle = new SessionHandle(null, sessionHandlePtr);
+            var session = new Session(handle);
+            var messageString = message.AsString();
 
-                SessionException exception;
+            SessionException exception;
 
-                if (isClientReset)
-                {
-                    var userInfo = StringStringPair.UnmarshalDictionary(userInfoPairs, userInfoPairsLength.ToInt32());
-                    exception = new ClientResetException(session.User.App, messageString, errorCode, userInfo);
-                }
-                else if (errorCode == ErrorCode.PermissionDenied)
+            if (isClientReset)
+            {
+                var userInfo = StringStringPair.UnmarshalDictionary(userInfoPairs, userInfoPairsLength.ToInt32());
+                exception = new ClientResetException(session.User.App, handle, messageString, errorCode, userInfo);
+            }
+            else if (errorCode == ErrorCode.PermissionDenied)
+            {
+                var userInfo = StringStringPair.UnmarshalDictionary(userInfoPairs, userInfoPairsLength.ToInt32());
+                exception = new PermissionDeniedException(session.User.App, messageString, userInfo);
+            }
+            else
+            {
+                exception = new SessionException(messageString, errorCode);
+            }
+
+            System.Threading.ThreadPool.QueueUserWorkItem(_ =>
+            {
+                try
                 {
-                    var userInfo = StringStringPair.UnmarshalDictionary(userInfoPairs, userInfoPairsLength.ToInt32());
-                    exception = new PermissionDeniedException(session.User.App, messageString, userInfo);
+                    Session.RaiseError(session, exception);
                 }
-                else
+                catch (Exception ex)
                 {
-                    exception = new SessionException(messageString, errorCode);
+                    Logger.Default.Log(LogLevel.Warn, $"An error has occurred while handling a session error: {ex}");
                 }
```

Comment:
We still need a `finally` block here that will dispose the session handle we created earlier.

Reply:
The handle is being released in `ClientResetException` when `InitiateClientReset` is invoked

Reply:
What if the user doesn't call `InitiateClientReset` or the exception isn't `ClientResetException` to begin with?

Reply:
We have a problem 😁